tests/drives_tests.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2020-2021  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+/* This sample shows how to write a simple unit test for dosbox-staging using
+ * Google C++ testing framework.
+ *
+ * Read Google Test Primer for reference of most available features, macros,
+ * and guidance about writing unit tests:
+ *
+ * https://github.com/google/googletest/blob/master/googletest/docs/primer.md#googletest-primer
+ */
+
+/* Include necessary header files; order of headers should be as follows:
+ *
+ * 1. Header declaring functions/classes being tested
+ * 2. <gtest/gtest.h>, which declares the testing framework
+ * 3. Additional system headers (if needed)
+ * 4. Additional dosbox-staging headers (if needed)
+ */
+
+#include "drives.h"
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+std::string run_Set_Label(char const * const input, bool cdrom) {
+    char output[32] = { 0 };
+    Set_Label(input, output, cdrom);
+    std::cout << "Set_Label " << "CD-ROM? " << (cdrom ? 'y' : 'n') << \
```

Comment:
Is this cout << part of the testing (or are we constructing something?)
Or do we want this info printed in the testing output?

We generally prefer printf or LOG_MSG(..) with printf formatting (c++20 eventually gives us format strings).

``` c++
printf("Set_Label CDROM? %c Input: %s Output: %s\n", 
       cdrom ? 'y' : 'n', input, output);
```